projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/execution/JobExecutionLogsUrlBuilder.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2021 VMware, Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.vmware.taurus.service.execution;
+
+import java.time.OffsetDateTime;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.text.StringSubstitutor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import com.vmware.taurus.service.model.DataJobExecution;
+
+/**
+ * Builds logs URL for each data job execution by specified template
+ * (datajobs.executions.logsUrl.template).
+ *
+ * Supported variables which will be replaced in the template
+ * with the particular execution values:
+ *
+ * <ul>
+ *    <li>{@value VAR_PREFIX}{@value EXECUTION_ID_VAR}{@value VAR_SUFFIX}</li>
+ *    <li>{@value VAR_PREFIX}{@value OP_ID_VAR}{@value VAR_SUFFIX}</li>
+ *    <li>{@value VAR_PREFIX}{@value JOB_NAME_VAR}{@value VAR_SUFFIX}</li>
+ *    <li>{@value VAR_PREFIX}{@value START_TIME_VAR}{@value VAR_SUFFIX}</li>
+ *    <li>{@value VAR_PREFIX}{@value END_TIME_VAR}{@value VAR_SUFFIX}</li>
+ * </ul>
+ *
+ * Example: "https://log-insight-url/li/query/stream?query=%C2%A7%C2%A7%C2%A7AND%C2%A7%C2%A7%C2%A7%C2%
+ * A7{{start_time}}%C2%A7{{end_time}}%C2%A7true%C2%A7COUNT%C2%A7text:CONTAINS:{{execution_id}}*"
+ */
+@Slf4j
+@Component
+public class JobExecutionLogsUrlBuilder {
+
+   // Default for testing purposes
+   static final String VAR_PREFIX = "{{";
+   static final String VAR_SUFFIX = "}}";
+
+   static final String EXECUTION_ID_VAR = "execution_id";
+   static final String OP_ID_VAR = "op_id";
+   static final String JOB_NAME_VAR = "job_name";
+   static final String START_TIME_VAR = "start_time";
+   static final String END_TIME_VAR = "end_time";
+
+   static final String ISO_DATE_FORMAT = "iso";
+   static final String UNIX_DATE_FORMAT = "unix";
+
+   @Value("${datajobs.executions.logsUrl.template}")
+   private String template;
+
+   @Value("${datajobs.executions.logsUrl.dateFormat}")
+   private String dateFormat;
+
+   public String build(DataJobExecution dataJobExecution) {
+      if (StringUtils.isEmpty(template)) {
+         log.warn("The property 'datajobs.executions.logsUrl.template' is empty!");
+         return "";
```

Comment:
nit: Maybe consider returning Optional ?

Reply:
I will think about that.